backend/api/users.py

```diff
@@ -57,41 +75,59 @@ def create_user():
 
 
 @users.route('/api/users', methods=['PUT'])
-@roles_required('admin')
+@login_required
 def update_user():
     user = request.get_json()
+    if(is_permitted(current_user, user)):
+        res = g.user_datastore.get_user(user['email'])
+        if res is None:
+            return make_response("Unknown User with Email-address: " +
+                                 user['email'], 400)
+        res.first_name = user['first name']
+        res.last_name = user['last name']
+        res.bio = user['bio']
+        res.save()
+        res = make_response(dumps(res))
+        res.headers['Content-Type'] = 'application/json'
 
-    res = g.user_datastore.get_user(user['email'])
-    if res is None:
-        return make_response("Unknown User with Email-address: " +
-                             user['email'], 400)
-    res.first_name = user['first name']
-    res.last_name = user['last name']
-    res.bio = user['bio']
-    res.save()
-    res = make_response(dumps(res))
-    res.headers['Content-Type'] = 'application/json'
+        return make_response("User with email: " +
+                             user['email'] + " updated", 200)
 
-    return make_response("User with email: " + user['email'] + " updated", 200)
+    return make_response("You don't have the permissions " +
+                         "to edit this user", 400)
 
 
 @users.route('/api/users/password', methods=['PUT'])
-@roles_required('admin')
+@login_required
 def update_password():
     user = request.get_json()
-
-    res = g.user_datastore.get_user(user['email'])
-    if res is None:
-        return make_response("Unknown User with Email-address: " +
-                             user['email'], 404)
-    old_password = user["old password"]
-    if verify_password(old_password, res.password):
+    if current_user["email"] == user['email']:
+        res = g.user_datastore.get_user(user['email'])
+        if res is None:
+            return make_response("Unknown User with Email-address: " +
+                                 user['email'], 404)
+        old_password = user["old password"]
+        if verify_password(old_password, res.password):
+            new_password = user["new password"]
+            if new_password == old_password:
+                return make_response("The old and new passwords" +
+                                     "can not be the same", 200)
+            res.password = encrypt_password(new_password)
+            res.save()
+            return make_response("Password updated!", 200)
+        return make_response("Old password is wrong", 400)
+    if current_user.has_role('admin'):
```

Comment:
I would rather not write the same code twice but just solve it like I suggested above